ocf_data_sampler/numpy_sample/nwp.py

```diff
@@ -0,0 +1,43 @@
+"""Convert NWP to NumpySample"""
+
+import pandas as pd
+import xarray as xr
+
+
+class NWPSampleKey:
+
+    nwp = 'nwp'
+    channel_names = 'nwp_channel_names'
+    init_time_utc = 'nwp_init_time_utc'
+    step = 'nwp_step'
+    target_time_utc = 'nwp_target_time_utc'
+    t0_idx = 'nwp_t0_idx'
+    y_osgb = 'nwp_y_osgb'
+    x_osgb = 'nwp_x_osgb'
+
+
+def convert_nwp_to_numpy_sample(da: xr.DataArray, t0_idx: int | None = None) -> dict:
```

Comment:
Potential improvement of function example as follows - purely considering the 'TO DO' comment and related modularity there primarily. 

```python

def convert_nwp_to_numpy_sample(da: xr.DataArray, t0_idx: int | None = None) -> dict:
    """Convert from Xarray to NWP NumpySample"""
    
    # Create example and add t if available
    example = create_data(da)
    
    if "target_time_utc" in da.coords:
        example[NWPSampleKey.target_time_utc] = da.target_time_utc.values.astype(float)
    
    # Add OSGB coordinates if available and / or necessary
    add_osgb(example, da)
    
    if t0_idx is not None:
        example[NWPSampleKey.t0_idx] = t0_idx
        
    return example


def create_data(da: xr.DataArray) -> dict:
    return {
        NWPSampleKey.nwp: da.values,
        NWPSampleKey.channel_names: da.channel.values,
        NWPSampleKey.init_time_utc: da.init_time_utc.values.astype(float),
        NWPSampleKey.step: (da.step.values / pd.Timedelta("1h")).astype(int),
    }


def add_osgb(example: dict, da: xr.DataArray) -> None:
    osgb_coords = (
        (NWPSampleKey.y_osgb, "y_osgb"),
        (NWPSampleKey.x_osgb, "x_osgb"),
    )
    
    for sample_key, dataset_key in osgb_coords:
        if dataset_key in da.coords:
            example[sample_key] = da[dataset_key].values

```